cmake/HoloHubConfigHelpers.cmake

```
@@ -13,6 +13,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Helper function to build packages
+function(add_holohub_package NAME)
+  set(pkgname "PKG_${NAME}")
+  option(${pkgname} "Build the ${NAME} package" ${BUILD_ALL})
+
+  message(DEBUG "${pkgname} = ${${pkgname}}")
+
+  # Configure the package if enabled
+  if(NOT ${pkgname})
+    return()
+  endif()
+  add_subdirectory(${NAME})
+
+  # If we have dependencies make sure they are built
+  cmake_parse_arguments(DEPS "" "" "EXTENSIONS;OPERATORS;APPLICATIONS" ${ARGN})
+  message(DEBUG "${pkgname} exts = ${DEPS_EXTENSIONS}")
+  message(DEBUG "${pkgname} ops = ${DEPS_OPERATORS}")
+  message(DEBUG "${pkgname} apps = ${DEPS_APPLICATIONS}")
+  foreach(dep IN LISTS DEPS_EXTENSIONS)
+    set("EXT_${dep}" ON CACHE BOOL "Build the ${dep} GXF extension" FORCE)
+  endforeach()
+  foreach(dep IN LISTS DEPS_OPERATORS)
+    set("OP_${dep}" ON CACHE BOOL "Build the ${dep} holoscan operator" FORCE)
+  endforeach()
+  foreach(dep IN LISTS DEPS_APPLICATIONS)
+    set("APP_${dep}" ON CACHE BOOL "Build the ${dep} application" FORCE)
+  endforeach()
```

Seems like this block should be conditional on the package being enabled, otherwise every HoloHub build will always enable these options. See https://github.com/nvidia-holoscan/holohub/blob/main/cmake/HoloHubConfigHelpers.cmake#L28

```suggestion
if(PKG_${NAME})
  foreach(dep IN LISTS DEPS_EXTENSIONS)
    set("EXT_${dep}" ON CACHE BOOL "Build the ${dep} GXF extension" FORCE)
  endforeach()
  foreach(dep IN LISTS DEPS_OPERATORS)
    set("OP_${dep}" ON CACHE BOOL "Build the ${dep} holoscan operator" FORCE)
  endforeach()
  foreach(dep IN LISTS DEPS_APPLICATIONS)
    set("APP_${dep}" ON CACHE BOOL "Build the ${dep} application" FORCE)
  endforeach()
endif()
```

---

It's done here: https://github.com/nvidia-holoscan/holohub/pull/637/files/d884c75abdc479a55121ab36a7855e07505805e8#diff-6771461c01c7c17ac1f47ce83a973b8923fce6ba006db2cbc0061b04abd1d858R24-R26

I just find the "return early" approach more elegant, and it also lowers the level of nesting below.

---

Ah I missed that, thanks!